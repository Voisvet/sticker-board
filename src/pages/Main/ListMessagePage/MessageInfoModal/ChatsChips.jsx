import React from 'react';
import Chip from '@material-ui/core/Chip';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  chip: {
    margin: 0.5 * theme.spacing.unit
  }
});

const ChatsChips = (props) => {
  // Display three first chats
  return props.chats.map(chat => (
    <Chip
      key={chat}
      color="primary"
      label={chat}
      className={props.classes.chip}
    />
  ));
};

export default withStyles(styles)(ChatsChips);
