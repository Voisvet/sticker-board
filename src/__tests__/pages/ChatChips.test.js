import React from 'react';
import renderer from 'react-test-renderer';
import ChatsChips from '../../pages/Main/ListMessagePage/MessageInfoModal/ChatsChips';

test("Chat chips render correctly", () => {
  const testData = ["Chat name 1", "Chat name 2", "Another chat name..."];

  const component = renderer.create(<ChatsChips chats={testData}/>).toJSON();
  expect(component).toMatchSnapshot();
});
