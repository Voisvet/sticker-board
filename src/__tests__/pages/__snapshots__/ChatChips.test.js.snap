// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat chips render correctly 1`] = `
Array [
  <div
    className="MuiChip-root-2 MuiChip-colorPrimary-3 ChatsChips-chip-1"
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={-1}
  >
    <span
      className="MuiChip-label-21"
    >
      Chat name 1
    </span>
  </div>,
  <div
    className="MuiChip-root-2 MuiChip-colorPrimary-3 ChatsChips-chip-1"
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={-1}
  >
    <span
      className="MuiChip-label-21"
    >
      Chat name 2
    </span>
  </div>,
  <div
    className="MuiChip-root-2 MuiChip-colorPrimary-3 ChatsChips-chip-1"
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={-1}
  >
    <span
      className="MuiChip-label-21"
    >
      Another chat name...
    </span>
  </div>,
]
`;
